test/lit/passes/remove-unused-brs-gc.wast

```diff
@@ -293,26 +616,34 @@
  ;; CHECK-NEXT:   )
  ;; CHECK-NEXT:  )
  ;; CHECK-NEXT:  (drop
- ;; CHECK-NEXT:   (if (result anyref)
- ;; CHECK-NEXT:    (local.get $x)
- ;; CHECK-NEXT:    (block $something (result anyref)
- ;; CHECK-NEXT:     (drop
- ;; CHECK-NEXT:      (br_on_cast $something (ref null $struct) (ref $struct)
- ;; CHECK-NEXT:       (local.get $struct)
+ ;; CHECK-NEXT:   (select (result (ref null $struct))
+ ;; CHECK-NEXT:    (block (result (ref null $struct))
+ ;; CHECK-NEXT:     (block $something (result (ref null $struct))
+ ;; CHECK-NEXT:      (drop
+ ;; CHECK-NEXT:       (block (result nullref)
+ ;; CHECK-NEXT:        (br_on_non_null $something
+ ;; CHECK-NEXT:         (local.get $struct)
+ ;; CHECK-NEXT:        )
+ ;; CHECK-NEXT:        (ref.null none)
+ ;; CHECK-NEXT:       )
```

Comment:
I understand why the `br_on_cast` in this existing test turned into `br_on_non_null + null`, but why did the `if` get selectified on new line 619? I guess because we consider a `br_on_non_null` to be much less work than `br_on_cast`?

Reply:
Looks like that's the case in `cost.h` - we consider casts very expensive ("infinity"), but null check pretty cheap ("2").

Reply:
Yep, that sounds right. Do you think there's anything worth changing here?

Reply:
Ah, no, this looks right.